include/solvers/mpm_base.tcc

```diff
@@ -96,10 +96,44 @@ mpm::MPMBase<Tdim>::MPMBase(const std::shared_ptr<IO>& io) : mpm::MPM(io) {
   try {
     if (post_process_.at("vtk_statevars").is_array() &&
         post_process_.at("vtk_statevars").size() > 0) {
-      for (unsigned i = 0; i < post_process_.at("vtk_statevars").size(); ++i) {
-        std::string attribute =
-            post_process_["vtk_statevars"][i].template get<std::string>();
-        vtk_statevars_.emplace_back(attribute);
+      // Iterate over state_vars
+      for (const auto& svars : post_process_["vtk_statevars"]) {
+        // Initial phase id and state vars
+        unsigned phase_id = 0;
+        std::vector<std::string> state_var;
+
+        // If state variables are arranged as string
+        if (svars.is_string()) {
+          std::string attribute = svars.template get<std::string>();
+          state_var.emplace_back(attribute);
+        }
+        // If state variables are arranged as json object
+        else if (svars.is_object()) {
+          // Phase id
+          if (svars.contains("phase_id"))
+            phase_id = svars.at("phase_id").template get<unsigned>();
+
```

Comment:
Should we add a check for the valid input of phase_id?

Reply:
Good point. I think this part is defined globally for simulating single-phase and two-phase, so checking `phase_id` might be a bit difficult as we need to go to particle level initially unless we specify it somewhere. Nevertheless, if you look at these lines:
https://github.com/cb-geo/mpm/blob/3b20da5fcace5ed390c8ff58b08f0040fa27d51e/include/particles/particle.h#L241-L247
Maybe we can use `state_variables_.at(phase).find(var)` instead of `state_variables_[phase].find(var)` to assert if the given phase is not allocated. What do you think @kks32?

Reply:
Any further thoughts on this?

Reply:
Doesn't this guarantee that the variable is available: `state_variables_[phase].find(var) != state_variables_[phase].end()`

Reply:
Yes, if the `state_variables_` have the appropriate `phase`. If the passed `phase` is equal or larger than `state_variables_.size()`, then it will cause a segmentation fault. For instance, if I am accessing index `phase=1` in single-phase particle, it will be seg-fault

Reply:
@kks32 Are we good with this? The current implementation has no `phase_id` check. So if input `phase_id` is not available, we will have seg fault.